### PR TITLE
[WIP]  provider/aws: Add JSON validation to the aws_vpc_endpoint resource.

### PR DESCRIPTION
**Pull request by kwilczynski**
This commit adds support for new helper function which is used to
normalise and validate JSON string.

Signed-off-by: Krzysztof Wilczynski <krzysztof.wilczynski@linux.com>

### CODE REVIEW

**Comment by kwilczynski** [comment]
Part of work which began in #8028.

**Comment by kwilczynski** [comment]
Acceptance test is passing:

```
$ make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSVpcEndpoint_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/09/14 23:29:08 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSVpcEndpoint_ -timeout 120m
=== RUN   TestAccAWSVpcEndpoint_importBasic
--- PASS: TestAccAWSVpcEndpoint_importBasic (55.89s)
=== RUN   TestAccAWSVpcEndpoint_basic
--- PASS: TestAccAWSVpcEndpoint_basic (56.15s)
=== RUN   TestAccAWSVpcEndpoint_withRouteTableAndPolicy
--- PASS: TestAccAWSVpcEndpoint_withRouteTableAndPolicy (107.48s)
PASS
ok     	github.com/hashicorp/terraform/builtin/providers/aws   	219.544s
```

File: builtin/providers/aws/resource_aws_vpc_endpoint.go [hunk 1]
```diff
@@ -128,8 +132,10 @@ func resourceAwsVPCEndpointRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("There are multiple prefix lists associated with the service name '%s'. Unexpected", prefixListServiceName)
 	}
 
+	policy, _ := normalizeJsonString(*vpce.PolicyDocument)
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_vpc_endpoint.go]: Is there *any* chance we need to check for an error here?

**Comment by stack72** [comment]
LGTM! Thanks for the work here :)

**Pull request merged**